.NET/Microsoft.Recognizers.Text.Sequence/SequenceRecognizer.cs

```diff
@@ -1,32 +1,67 @@
-﻿using System;
+﻿using Microsoft.Recognizers.Text.Sequence.English;
+using System;
 using System.Collections.Generic;
 
-using Microsoft.Recognizers.Text.Sequence.English;
-
 namespace Microsoft.Recognizers.Text.Sequence
 {
-    public class SequenceRecognizer : Recognizer
+    public class SequenceRecognizer : Recognizer<SequenceOptions>
     {
-        public static readonly SequenceRecognizer Instance = new SequenceRecognizer(SequenceOptions.None);
+        public SequenceRecognizer(string targetCulture, SequenceOptions options = SequenceOptions.None, bool lazyInitialization = false)
+            : base(targetCulture, options, lazyInitialization)
+        {
+        }
+
+        public SequenceRecognizer(string targetCulture, int options, bool lazyInitialization = false)
+            : this(targetCulture, GetOption(options), lazyInitialization)
+        {
+        }
 
-        private SequenceRecognizer(SequenceOptions options)
+        public SequenceRecognizer(SequenceOptions options = SequenceOptions.None, bool lazyInitialization = true)
+            : base(null, options, lazyInitialization)
         {
-            RegisterModel(Culture.English, options.ToString(), new Dictionary<Type, IModel>
-            {
-                [typeof(PhoneNumberModel)] = new PhoneNumberModel(new PhoneNumberParser(), new PhoneNumberExtractor()),
-                [typeof(IpAddressModel)] = new IpAddressModel(new IpParser(), new IpExtractor())
-            });
         }
 
-        public IModel GetPhoneNumberModel(string culture, bool fallbackToDefaultCulture = true)
+        public SequenceRecognizer(int options, bool lazyInitialization = true)
+            : this(null, GetOption(options), lazyInitialization)
         {
-            return GetModel<PhoneNumberModel>(Culture.English, fallbackToDefaultCulture, SequenceOptions.None.ToString());
         }
 
-        public IModel GetIpAddressModel(string culture, bool fallbackToDefaultCulture = true)
+        public IModel GetPhoneNumberModel(string culture = null, bool fallbackToDefaultCulture = true)
         {
-            return GetModel<IpAddressModel>(Culture.English, fallbackToDefaultCulture, SequenceOptions.None.ToString());
+            return GetModel<PhoneNumberModel>(culture, fallbackToDefaultCulture);
         }
 
+        public IModel GetIpAddressModel(string culture = null, bool fallbackToDefaultCulture = true)
+        {
+            return GetModel<IpAddressModel>(culture, fallbackToDefaultCulture);
```

Comment:
We've left the method arguments and hardcoded to ` Culture.English` . For PhoneNumber, we'll probably add new cultures in the future. But for IPAddress, should we simply remove the culture and fallback arguments since there won't be a culture-specific IP Address?

```C#
public IModel GetIpAddressModel(string culture = null, bool fallbackToDefaultCulture = true)
{
    return GetModel<IpAddressModel>(Culture.English, fallbackToDefaultCulture);
}
```